pkg/controller/core/traits/manualscalertrait/manualscalertrait_controller.go

```diff
@@ -133,52 +124,43 @@ func (r *ManualScalerTraitReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		Controller:         &isController,
 		BlockOwnerDeletion: &bod,
 	}
-
-	existingRefs := scaleDeploy.GetOwnerReferences()
-	fi := -1
-	for i, r := range existingRefs {
-		if r.UID == manualScaler.UID {
-			fi = i
-			break
+	for _, res := range resources {
```

Comment:
resources is a concept of an intermediate process, It shouldn't be in Reconcile method,
Reconcile method just call several functions/method with their own standard input and output
```
func (itb *ManualScalerTraitBuilder) doManualScaler(ctx context.Context, workload unstructured.Unstructured, trait *corev1alpha2.ManualScalerTrait) (oam.Object, error) {
```

Reply:
> resources is a concept of an intermediate process, It shouldn't be in Reconcile method,

I am fine with some refactoring but I am not sure what this means. What can be in the Reconcile method then? It seems that everything is an intermediate process.

Reply:
I didn't say it correctly.
method of Reconcile  is the controller's entrance
so it should call functions instead of writing business code directly
it's very clear and easy to read,
What arguments does that function take, what arguments does it return, and function name tells you what it's going to do, just like your refactoring, excellent job
```
func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
	ctx := context.Background()
	var manualScalar oamv1alpha2.ManualScalerTrait
	i r.Get(ctx, req.NamespacedName, &manualScalar)
	// Fetch the workload instance this trait is referring to
	workload, result, err := r.fetchWorkload(ctx, &manualScalar)
	// Fetch the child resources list from the corresponding workload
	resources, err := util.FetchWorkloadDefinition(ctx, r, workload)
	// Scale the child resources we know how to scale
	result, err = r.scaleChildResources(ctx, manualScalar, resources)
```
workload, trait are OAM concept, they appear here, the rest may can be encapsulated too
```
func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
	ctx := context.Background()
	var manualScalar oamv1alpha2.ManualScalerTrait
	i r.Get(ctx, req.NamespacedName, &manualScalar)
	// Fetch the workload instance this trait is referring to
	workload, err := r.fetchWorkload(ctx, &manualScalar)
	// do manual scaler by workload and trait
        err := r.doManualScaler(ctx, workload,manualScalar)
```